include/audit_events.hpp

```diff
@@ -123,76 +123,157 @@ inline bool wantAudit(const crow::Request& req)
+        ((req.method() == boost::beast::http::verb::post) &&
+         checkPostUser(req)))
+    {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * @brief Checks if request's detail data should be logged
+ *
+ * @return True if request's detail data should be logged
+ */
+inline bool wantDetail(const crow::Request& req)
 {
-    int code = __LINE__;
+    switch (req.method())
+    {
+        case boost::beast::http::verb::patch:
+        case boost::beast::http::verb::post:
+            if (checkSkipDetail(req))
+            {
+                return false;
+            }
+            return true;
+
+        case boost::beast::http::verb::put:
+            return (!req.target().starts_with("/ibm/v1"));
+
+        case boost::beast::http::verb::delete_:
+            return true;
+
+        default:
+            // Shouldn't be here, don't log any data
+            BMCWEB_LOG_DEBUG << "Unexpected verb " << req.method();
+            return false;
+    }
+}
 
+inline void auditEvent(const crow::Request& req, const std::string& userName,
+                       bool success)
+{
+    std::string opPath;
     char cnfgBuff[256];
     size_t bufLeft = 256; // Amount left available in cnfgBuff
     char* user = NULL;
     size_t opPathLen;
     size_t userLen = 0;
     int rc;
     int origErrno;
+    std::string ipAddress;
+    std::string detail = "";
 
     if (auditOpen() == false)
     {
         return;
     }
 
-    opPathLen = std::strlen(opPath) + 1;
+    opPath = "op=" + std::string(req.methodString()) + ":" +
+             std::string(req.target()) + " ";
+    // Account for NULL
+    opPathLen = opPath.length() + 1;
     if (opPathLen > bufLeft)
     {
         // Truncate event message to fit into fixed sized buffer.
         BMCWEB_LOG_WARNING << "Audit buffer too small, truncating:"
                            << " bufLeft=" << bufLeft
                            << " opPathLen=" << opPathLen;
         opPathLen = bufLeft;
-        code = __LINE__;
     }
-    strncpy(cnfgBuff, opPath, opPathLen);
+    strncpy(cnfgBuff, opPath.c_str(), opPathLen);
     cnfgBuff[opPathLen - 1] = '\0';
     bufLeft -= opPathLen;
 
+    // Determine any additional info for the event
+    if (wantDetail(req))
+    {
+        detail = req.body + " ";
+    }
+
+    if (detail.length() > 0)
+    {
+        if (detail.length() > bufLeft)
+        {
+            // Additional info won't fix into fixed sized buffer. Leave
+            // it off.
+            BMCWEB_LOG_WARNING << "Audit buffer too small for data:"
+                               << " bufLeft=" << bufLeft
+                               << " detailLen=" << detail.length();
+        }
+        else
+        {
+            strncat(cnfgBuff, detail.c_str(), detail.length());
+            bufLeft -= detail.length();
+        }
+    }
+
     // encode user account name to ensure it is in an appropriate format
     user = audit_encode_nv_string("acct", userName.c_str(), 0);
```

Comment:
Similar,  the variables can defined here.
```
    char* user = audit_encode_nv_string("acct", userName.c_str(), 0);
    size_t userLen = 0;
```

Reply:
Either one should be okay in this case as the only one and here is accessing it.